controllers/openstackdataplanenodeset_controller.go

```diff
@@ -384,6 +384,24 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
 		}
 	}
 
+	isDeploymentReady, isDeploymentRunning, isDeploymentFailed, deploymentRequeueTime, err := checkDeployment(helper, instance)
+	if err != nil {
+		instance.Status.Conditions.MarkFalse(
+			condition.DeploymentReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityError,
+			condition.DeploymentReadyErrorMessage,
+			err.Error())
+		Log.Error(err, "Unable to get deployed OpenStackDataPlaneDeployments.")
+		return ctrl.Result{}, err
+	}
+
+	if isDeploymentRunning {
+		Log.Info("Deployment running, requeueing...", "instance", instance)
+		return ctrl.Result{RequeueAfter: time.Second * time.Duration(deploymentRequeueTime)}, nil
```

Comment:
Just for record, as I mentioned earlier, this `deploymentRequeueTime` default of 15 secs could make sense when deployment has not started deploying services but probably not when a deployment is already deploying services.

Reply:
Given that the NodeSet is already watching the Deployment, I don't think we need to explicitly requeue here at all.